src/engine/Drawing/SpriteSheet.ts

```diff
@@ -34,28 +35,28 @@ export class SpriteSheetImpl {
     rows?: number,
     spWidth?: number,
     spHeight?: number,
-    spacing?: number
+    spacing?: number | SpriteSheetSpacingDimensions
   ) {
     let loadFromImage: boolean = false;
     if (imageOrConfigOrSprites instanceof Array) {
-      this.sprites = imageOrConfigOrSprites;
+      this._sprites = imageOrConfigOrSprites;
     } else {
       if (imageOrConfigOrSprites && !(imageOrConfigOrSprites instanceof Texture)) {
-        this.columns = imageOrConfigOrSprites.columns;
-        this.rows = imageOrConfigOrSprites.rows;
-        this.spWidth = imageOrConfigOrSprites.spWidth;
-        this.spHeight = imageOrConfigOrSprites.spHeight;
-        this.image = imageOrConfigOrSprites.image;
-        this.spacing = imageOrConfigOrSprites.spacing || 0;
+        this._columns = imageOrConfigOrSprites.columns;
+        this._rows = imageOrConfigOrSprites.rows;
+        this._spWidth = imageOrConfigOrSprites.spWidth;
+        this._spHeight = imageOrConfigOrSprites.spHeight;
+        this._image = imageOrConfigOrSprites.image;
+        this._spacing = imageOrConfigOrSprites.spacing || 0;
       } else {
-        this.image = <Texture>imageOrConfigOrSprites;
-        this.columns = columns;
-        this.rows = rows;
-        this.spWidth = spWidth;
-        this.spHeight = spHeight;
-        this.spacing = spacing || 0;
+        this._image = <Texture>imageOrConfigOrSprites;
+        this._columns = columns;
+        this._rows = rows;
+        this._spWidth = spWidth;
+        this._spHeight = spHeight;
+        this._spacing = spacing || 0;
       }
-      this.sprites = new Array(this.columns * this.rows);
+      this._sprites = new Array(this._columns * this._rows);
       loadFromImage = true;
     }
```

Comment:
After this initialization routine, the rest of the code in ctor uses public getters for consistency.